src/logic/game.cc

```diff
@@ -217,26 +217,16 @@ void Game::save_syncstream(bool const save) {
 	syncwrapper_.syncstreamsave_ = save;
 }
 
-void Game::check_addons_desync_magic() {
-	if (did_check_addons_desync_magic_) {
-		postload_addons();
+void Game::postload_addons_before_loading() {
+	if (did_postload_addons_before_loading_) {
 		return;
 	}
-	did_check_addons_desync_magic_ = true;
-
-	// TODO(Nordfriese): Minimal-invasive fix for a very evil desync. The problem is:
-	// - We load only the tribes actually being played when starting a game.
-	// - With some add-ons, it is necessary to load *all* tribes during loading the replay.
-	// - This results in different load orders, so the same item may have a different
-	//   DescriptionIndex in the replay than in the original game. In particular, it
-	//   may happen that the index of A is smaller than B's in the original game but
-	//   greater than B's in the replay.
-	// The simple (and ugly) solution is to load all tribes in the original game if
-	// we know that we'll need to do this in the replay as well. We thus get the
-	// same load order – at the cost of a much longer loading time, so this is not
-	// suited as a long-term solution. In the long run, it would probably be best
-	// to get rid of `DescriptionIndex` entirely in favour of `std::string`.
+	did_postload_addons_before_loading_ = true;
+	delete_world_and_tribes();
```

Comment:
// We can safely delete because ...
- I assume this is some data needed during loading only

Reply:
This is still open?

Reply:
I have deleted the old function `check_addons_desync_magic` because, as the name suggests, it was just a workaround for a desync. These lines belong to a new function `postload_addons_before_loading` which prepares a game's descriptions to load new data during saveloading. The git diff is merging the deleted and the added function because they happen to have some lines in common, but there's no actual relation here.